common/src/main/java/org/wso2/testgrid/common/util/tinkerer/ScriptExecutorThread.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.common.util.tinkerer;
+
+import com.google.gson.Gson;
+import org.glassfish.jersey.client.ChunkedInput;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.agentoperation.OperationSegment;
+import org.wso2.testgrid.common.exception.TestGridException;
+import org.wso2.testgrid.common.util.FileUtil;
+import org.wso2.testgrid.common.util.StringUtil;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.Response;
+
+/**
+ * Thread to stream data from tinkerer and dump result into a file.
+ */
+public class ScriptExecutorThread extends Thread {
+
+    private static final Logger logger = LoggerFactory.getLogger(ScriptExecutorThread.class);
+
+    public static final int MAX_BUFFER_IDLE_TIME = 900000;  // Maximum waiting time to update message queue
```

Comment:
is this in milliseconds? The field name need to reflect that distinction.

Reply:
@madushadhanushka is this in milli seconds? If so I can edit the variable name.

Reply:
it is in milli seconds